### ISSUE TITLE
ReturnCount does not violate ctor with returns

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11212]
I have read check documentation: https://checkstyle.org/config_coding.html#ReturnCount
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
          <property name="max" value="0"/>
          <property name="tokens" value="CTOR_DEF"/>
        </module>
    </module>
</module>

/var/tmp $ cat Test.java
public class Test {
    public Test() {
      return;
    } // OK ???  should be violation
}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2.1-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

---

Violation is expected on ctor.

**Comment by strkkk** [comment]
@romani there is a property maxForVoid for void returns 
https://checkstyle.org/config_coding.html#ReturnCount_Properties

it reports violation for this code, so there is no bug.
```
$ cat Test.java
public class Test {
    public Test() {
      return;
    }
}

$ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">

    <module name="TreeWalker">
        <module name="ReturnCount">
         <property name="maxForVoid" value="0"/>
          <property name="tokens" value="CTOR_DEF"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-9.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\workdir\Test.java:2:5: Return count is 1 (max allowed for void methods/constructors/lambdas is 0). [ReturnCount]
Audit done.
Checkstyle ends with 1 errors.

```
